eva-accession-deprecate/src/main/java/uk/ac/ebi/eva/accession/deprecate/io/DeprecationWriter.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.deprecate.io;
+
+import org.springframework.batch.item.ItemWriter;
+import org.springframework.data.mongodb.core.BulkOperations;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import uk.ac.ebi.ampt2d.commons.accession.core.models.EventType;
+
+import uk.ac.ebi.eva.accession.core.persistence.DbsnpClusteredVariantEntity;
+import uk.ac.ebi.eva.accession.core.persistence.DbsnpClusteredVariantInactiveEntity;
+import uk.ac.ebi.eva.accession.core.persistence.DbsnpClusteredVariantOperationEntity;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class DeprecationWriter implements ItemWriter<DbsnpClusteredVariantEntity> {
+
+    private static final String DBSNP_CLUSTERED_VARIANT_DECLUSTERED_COLLECTION_NAME =
+            "dbsnpClusteredVariantEntityDeclustered";
+
+    private static final String ID_FIELD = "_id";
+
+    private MongoTemplate mongoTemplate;
+
+    public DeprecationWriter(MongoTemplate mongoTemplate) {
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    @Override
+    public void write(List<? extends DbsnpClusteredVariantEntity> deprecableClusteredVariants) throws Exception {
+        removeDeprecableClusteredVariants(deprecableClusteredVariants);
+        insertDeprecateOperation(deprecableClusteredVariants);
```

Comment:
after thinking again about this, I think I prefer to insert first. As we don't have transactions, if something fails with the current order, we might loose track of RSs that should be deprecated and they won't appear in the main RS collection nor in the RS operations collection, only in the SS operations.

If something fails while inserting first, we can end up with repeated RS operations, which is recoverable.

Reply:
Agree with writing first. This is also how moving between disk partitions work, first copying then deleting, to avoid losing any information.

Reply:
so one thing we can do is catch any exception  in this function and log the list of accessions, in case we have to recover the chunk manually